pandas/tests/extension/test_common.py

```diff
@@ -9,30 +12,81 @@
 from pandas.core.arrays import ExtensionArray
 
 
-class DummyDtype(dtypes.ExtensionDtype):
+class DummyClass:
     pass
 
 
+class DummyDtype(dtypes.ExtensionDtype):
+
+    type = DummyClass
+    name = "dummy"
+
+    @classmethod
+    def construct_from_string(cls, string):
+        if string == cls.name:
+            return cls()
+        else:
+            raise TypeError(f"Cannot construct a '{cls}' from '{string}'")
+
+    @classmethod
+    def construct_array_type(cls):
+        return DummyArray
+
+
 class DummyArray(ExtensionArray):
+
+    _dtype = DummyDtype
+
     def __init__(self, data):
-        self.data = data
+        self.data = np.array(data)
 
-    def __array__(self, dtype):
+    def __array__(self, dtype=None):
         return self.data
 
     @property
     def dtype(self):
-        return DummyDtype()
+        return DummyArray._dtype()
```

Comment:
why do you need the inderection of _dtype? what is wrong with what was here before

Reply:
It's because I'm subclassing DummyDtype and DummyArray in order to test several possible na_values

https://github.com/venaturum/pandas/blob/69117c1ebd1621a69164129169232c83a7439b19/pandas/tests/extension/test_common.py#L138

```
@pytest.mark.parametrize("na_value", [np.nan, pd.NA, None])
def test_empty_series_construction(na_value):
    class TempDType(DummyDtype):
        @classmethod
        def construct_array_type(cls):
            return TempArray

    TempDType.na_value = na_value

    class TempArray(DummyArray):
        _dtype = TempDType

    result = pd.Series(index=[1, 2, 3], dtype=TempDType())
    expected = pd.Series([na_value] * 3, index=[1, 2, 3], dtype=TempDType())
    tm.assert_series_equal(result, expected)
```